Description: Patch for CMakeLists.txt to generate correct libraries
 This patch will help generate shared library files with correct SONAME, 
 and generate appropriate static library file as well.
Author: Aron Xu <happyaron.xu@gmail.com>
Last-Update: 2011-07-09

--- libgooglepinyin-0.0+hg003.orig/src/CMakeLists.txt
+++ libgooglepinyin-0.0+hg003/src/CMakeLists.txt
@@ -1,3 +1,8 @@
+set(LIBGOOGLEPINYIN_API_VERSION "0.1.0")
+
+set(LIBGOOGLEPINYIN_ABI_VERSION "0.1.0")
+set(LIBGOOGLEPINYIN_ABI_MAJOR_VERSION "0")
+
 set(GOOGLEPINYIN_SRC
 	dictbuilder.cpp
 	dictlist.cpp
@@ -18,5 +23,20 @@ set(GOOGLEPINYIN_SRC
 )
 
 add_library(googlepinyin SHARED ${GOOGLEPINYIN_SRC})
-install(TARGETS googlepinyin DESTINATION ${libdir})
-target_link_libraries(googlepinyin pthread )
\ No newline at end of file
+target_link_libraries(googlepinyin pthread )
+set_target_properties(googlepinyin PROPERTIES
+    VERSION ${LIBGOOGLEPINYIN_ABI_VERSION}
+    SOVERSION ${LIBGOOGLEPINYIN_ABI_MAJOR_VERSION}
+    LINK_FLAGS ${CMAKE_SHARED_LINKER_FLAGS}
+)
+
+add_library(googlepinyin-static STATIC ${GOOGLEPINYIN_SRC})
+SET_TARGET_PROPERTIES(googlepinyin-static PROPERTIES
+    OUTPUT_NAME "googlepinyin"
+    PREFIX "lib"
+)
+
+install(TARGETS googlepinyin googlepinyin-static
+    LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+    ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
+)
